tests/Container/ContainerTest.php

```diff
@@ -66,6 +66,19 @@ public function testBindIfDoesRegisterIfServiceNotRegisteredYet()
         $this->assertSame('Dayle', $container->make('name'));
     }
 
+    public function testBindWhenIfConditionIsTrue()
+    {
+        $container = new Container;
+        $container->bind('surname', function () {
+            return 'Taylor';
+        });
+        $container->bindWhen(true, 'name', function () {
+            return 'Dayle';
+        });
```

Comment:
What about a test when condition is false?

Reply:
The class cannot be `bind'